### PR TITLE
Forbid to upload part files via web UI

### PR DESCRIPTION
**Pull request by MorrisJobke**
* downstream of https://github.com/owncloud/core/pull/27149

cc @nextcloud/javascript

### CODE REVIEW

**Comment by MorrisJobke** [comment]
I tested this and it works 👍

File: core/js/tests/specHelper.js [hunk 1]
```diff
@@ -94,7 +94,8 @@ window.oc_appswebroots = {
 };
 window.oc_config = {
 	session_lifetime: 600 * 1000,
-	session_keepalive: false
+	session_keepalive: false,
+	blacklist_files_regex: '\.(part|filepart)$',
```
Comment (ChristophWurst) [inline core/js/tests/specHelper.js]: Not a big fan of this global variable to be honest. Should be moved to https://github.com/nextcloud/server/pull/3950/files#diff-c493b80ad1a9bcfa6b69637097972371R232 unless we really want this to be configurable.

**Review by ChristophWurst — COMMENTED** [review]
Tested and works on FF, also with drag and drop

File: core/js/tests/specHelper.js (hunk 1 above)
Comment (MorrisJobke) [inline core/js/tests/specHelper.js]: I guess the idea is to use the same regex for backend and frontend 🤔

Comment (ChristophWurst) [inline core/js/tests/specHelper.js]: I see, so this is just the fallback/default value.

**Review by ChristophWurst — APPROVED** [review]

File: core/js/tests/specHelper.js (hunk 1 above)
Comment (MorrisJobke) [inline core/js/tests/specHelper.js]: No - this is the test spec :P

Comment (MorrisJobke) [inline core/js/tests/specHelper.js]: > No - this is the test spec :P

... because we don't run PHP in the JSUnit tests 😉

Comment (ChristophWurst) [inline core/js/tests/specHelper.js]: Gotcha!

File: apps/files/js/files.js [hunk 2]
```diff
@@ -101,7 +101,10 @@
 				throw t('files', '"{name}" is an invalid file name.', {name: name});
 			} else if (trimmedName.length === 0) {
 				throw t('files', 'File name cannot be empty.');
+			} else if (OC.fileIsBlacklisted(trimmedName)) {
+				throw t('files', '"{name}" is not an allow filetype', {name: name});
```
Comment (LukasReschke) [inline apps/files/js/files.js]: Should probably be `allowed`?

**Review by LukasReschke — CHANGES_REQUESTED** [review]

File: apps/files/js/files.js (hunk 2 above)
Comment (MorrisJobke) [inline apps/files/js/files.js]: Could you fix that?

**Comment by MorrisJobke** [comment]
Changes look good :+1:

**Review by LukasReschke — APPROVED** [review]

**Pull request merged**